.codegen/model.go.tmpl

```diff
@@ -18,20 +18,74 @@ import (
 	"github.com/databricks/databricks-sdk-go/marshal"
     "github.com/hashicorp/terraform-plugin-framework/types"
 )
+{{- $excluded := dict "ShareInfo"        (list "CreatedAt" "CreatedBy" "UpdatedAt" "UpdatedBy")
+                      "SharedDataObject" (list "AddedAt" "AddedBy" "Status") -}}
 {{range .Types}}
 {{- if or .Fields .IsEmpty}}
 {{.Comment "// " 80}}
 type {{.PascalName}} struct {
+    {{- $excluded := getOrDefault $excluded .PascalName (list) -}}
     {{- range .Fields}}
     {{.Comment "    // " 80}}
-    {{.PascalName}} {{template "type" .Entity}} `{{template "field-tag" . }}`{{end}}
+	{{- $data := dict "field" . "excluded" $excluded }}
+	{{template "field" $data}}{{if and .Entity.IsComputed (not (in $excluded .PascalName))}}{{ $data := dict "field" . "excluded" $excluded "effective" true }}{{printf "\n"}}{{template "field" $data}}{{end}}{{end}}
+}
+
+func (newState *{{.PascalName}}) SyncEffectiveFieldsDuringCreateOrUpdate(plan {{.PascalName}}) {
+  {{- range .Fields -}}
+  {{- if and .Entity.IsComputed (or .Entity.IsString .Entity.IsBool .Entity.IsInt64 .Entity.IsFloat64 .Entity.IsInt .Entity.Enum) -}}
+  {{- if not (in $excluded .PascalName) -}}
+  newState.Effective_{{.PascalName}} = newState.{{.PascalName}}
+  newState.{{.PascalName}} = plan.{{.PascalName}}
+  {{end}}
+  {{end}}
+  {{end}}
+}
+
+func (newState *{{.PascalName}}) SyncEffectiveFieldsDuringRead(existingState {{.PascalName}}) {
```

Comment:
For my understanding, where would we be calling these in the read? Asking because in case there is an error in the read, we shouldn't sync.

Reply:
`SyncEffectiveFieldsDuringRead` is a function that should be invoked before saving to the state during a Read operation, while `SyncEffectiveFieldsDuringCreateOrUpdate` should be called before saving to the state during a Create or Update operation. We might consider introducing an interface here to allow this syncing to be done automatically by the Resource. However, I would prefer to postpone this refactoring for the time being.